### PR TITLE
src: kw_env: Adding support to `kw env --no-env`

### PR DESCRIPTION
**Pull request by aquilamacedo**
This commit creates a mechanism allowing the user to "exit" the env resource using the `--no-env` parameter.

Closes: #746

Signed-off-by: Aquila Macedo <aquilamacedo@riseup.net>

### CODE REVIEW

**Comment by aquilamacedo** [comment]
Hi @rodrigosiqueira. I just created this draft issue #746. I would like to know if the implementation is according to your specifications.

File: documentation/man/features/env.rst [hunk 1]
```diff
@@ -48,6 +48,9 @@ OPTIONS
 -l, \--list:
   It shows all envs created via `\--create` option.
 
+-n, \--no-env:
+  Allow users to "exit" the env feature.
```
Comment (rodrigosiqueira) [inline documentation/man/features/env.rst:52]: After reading this, I think that `--exit-env` is a better name.

Comment (rodrigosiqueira) [reply documentation/man/features/env.rst:52]: Also, explain what exit an env means. Explain the behavior as we discussed in the https://github.com/kworkflow/kworkflow/issues/746 .

File: documentation/man/features/env.rst [hunk 2]
```diff
@@ -63,3 +66,7 @@ If you want to create a new env, you can use::
 If you want to list the available envs::
 
   kw env --list
+
+If you want to exit the env feature:::
```
Comment (rodrigosiqueira) [inline documentation/man/features/env.rst:70]: Delete one of the `:`

File: src/kw_env.sh [hunk 3]
```diff
@@ -72,6 +77,25 @@ function use_target_env()
   printf '%s\n' "$target_env" > "${local_kw_configs}/${ENV_CURRENT_FILE}"
 }
 
+function use_no_env()
```
Comment (rodrigosiqueira) [inline src/kw_env.sh:80]: add function documentation, and let's call this function `exit_env`.

File: src/kw_env.sh [hunk 4]
```diff
@@ -72,6 +77,25 @@ function use_target_env()
...
+{
+  local current_env
+  local local_kw_configs="${PWD}/.kw"
+
+  current_env=$(< "${local_kw_configs}/${ENV_CURRENT_FILE}")
+
+  if [[ $(ask_yN 'Do you want to make this env the default?') =~ '1' ]]; then
```
Comment (rodrigosiqueira) [inline src/kw_env.sh:87]: I think you can add a warning followed by the question, something like:
```
warning 'You are about to leave the env setup, and ENV_NAME config files will be used as a default.'
  if [[ $(ask_yN 'Do you really want to proceed?') =~ '1' ]]; then
```

File: src/kw_env.sh [hunk 5]
```diff
@@ -72,6 +77,25 @@ function use_target_env()
...
+    for config in "${config_file_list[@]}"; do
+      if [[ -L "${local_kw_configs}/${config}.config" ]]; then
```
Comment (rodrigosiqueira) [inline src/kw_env.sh:89]: Add a comment before this `if` that explains that you are removing the link but not the original file.

File: src/kw_env.sh [hunk 6]
```diff
@@ -72,6 +77,25 @@ function use_target_env()
...
+        rm "${local_kw_configs}/${config}.config"
+      fi
+      cp "${local_kw_configs}/${current_env}/${config}.config" "${local_kw_configs}"
+    done
+    printf '%s\n' 'NO ENV' > "${local_kw_configs}/${ENV_CURRENT_FILE}"
```
Comment (rodrigosiqueira) [inline src/kw_env.sh:94]: I think you can remove the current env file.

File: src/kw_env.sh [hunk 7]
```diff
@@ -72,6 +77,25 @@ function use_target_env()
...
+    success "You left the environment feature."
```
Comment (rodrigosiqueira) [inline src/kw_env.sh:95]: single quotes.

**Review by rodrigosiqueira — CHANGES_REQUESTED** [review]

**Comment by aquilamacedo** [comment]
Hi @rodrigosiqueira. I just finished making the requested changes :)

**Comment by rodrigosiqueira** [comment]
Thanks a lot for this PR. I already merged it with the following changes:

```bash
diff --git a/src/bash_autocomplete.sh b/src/bash_autocomplete.sh
index 64941a3..e5a01da 100644
--- a/src/bash_autocomplete.sh
+++ b/src/bash_autocomplete.sh
@@ -70,7 +70,7 @@ function _kw_autocomplete()
   kw_options['drm']='--remote --local --gui-on --gui-off --load-module
                      --unload-module --conn-available --modes --help'
 
-  kw_options['env']='--create --list --use'
+  kw_options['env']='--create --list --use --exit-env'
 
   kw_options['self-update']='--unstable --help'
   kw_options['u']="${kw_options['self-update']}"
diff --git a/src/kw_env.sh b/src/kw_env.sh
index ef42135..4163aa6 100644
--- a/src/kw_env.sh
+++ b/src/kw_env.sh
@@ -86,7 +86,7 @@ function exit_env()
 
   current_env=$(< "${local_kw_configs}/${ENV_CURRENT_FILE}")
 
-  warning 'You are about to leave the env setup, and ENV_NAME config files will be used as a default.'
+  warning "You are about to leave the env setup, and ${current_env} config files will be used as a default."
   if [[ $(ask_yN 'Do you really want to proceed?') =~ '1' ]]; then
     for config in "${config_file_list[@]}"; do
       # All symbolic links will be removed, and the current env configuration files
```

**Pull request closed without merging**